### PR TITLE
net: Add if_nameindex and if_freenameindex API

### PR DESCRIPTION
**Pull request by xiaoxiang781216**
## Summary
Specified here:
https://man7.org/linux/man-pages/man3/if_nameindex.3.html

## Impact
New API

## Testing
Internal test case.

### CODE REVIEW

**Review by Ouss4 — APPROVED** [review]

File: libs/libc/net/lib_nameindex.c [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+{
+  FAR struct if_nameindex *ifn;
+  FAR char *buf;
+  int i = 0;
+  int j;
+
+  ifn = lib_malloc((sizeof(*ifn) + IF_NAMESIZE) * MAX_IFINDEX);
+  if (ifn == NULL)
+    {
+      return NULL;
```
Comment (Ouss4) [inline libs/libc/net/lib_nameindex.c:66]: Should we set `errno` to `ENOBUFS` here?

Comment (xiaoxiang781216) [reply libs/libc/net/lib_nameindex.c:66]: Good point, but form spec, the errno should be set by malloc:
https://pubs.opengroup.org/onlinepubs/009695399/functions/malloc.html

Comment (Ouss4) [reply libs/libc/net/lib_nameindex.c:66]: https://pubs.opengroup.org/onlinepubs/9699919799/functions/if_nameindex.html says that `ENOBUFS` should be returned when insufficient resources are available, doesn't this mean "if malloc fails"?

Comment (xiaoxiang781216) [reply libs/libc/net/lib_nameindex.c:68]: Done, ENOBUFS is set explicitly.

**Review by Ouss4 — APPROVED** [review]

**Pull request merged**